exp/window_func.go

```diff
@@ -0,0 +1,106 @@
+package exp
+
+type sqlWindowFunctionExpression struct {
+	name       string
+	args       []interface{}
+	windowName string
+	window     WindowExpression
+}
+
+func NewSQLWindowFunctionExpression(name string, args ...interface{}) SQLWindowFunctionExpression {
+	return sqlWindowFunctionExpression{
+		name: name,
+		args: args,
+	}
+}
+
+func (swfe sqlWindowFunctionExpression) clone() sqlWindowFunctionExpression {
+	return sqlWindowFunctionExpression{
+		name:       swfe.name,
+		args:       swfe.args,
+		windowName: swfe.windowName,
+		window:     swfe.window,
+	}
+}
+
+func (swfe sqlWindowFunctionExpression) Clone() Expression {
+	return swfe.clone()
+}
+func (swfe sqlWindowFunctionExpression) Expression() Expression {
+	return swfe
+}
+func (swfe sqlWindowFunctionExpression) As(val interface{}) AliasedExpression {
+	return aliased(swfe, val)
+}
+func (swfe sqlWindowFunctionExpression) Eq(val interface{}) BooleanExpression  { return eq(swfe, val) }
```

Comment:
Are all of these valid comparisons for an SQLWindowFunction? If they are can you provide some examples of when you would use them otherwise I think we should remove them.

Reply:
like 
```sql
SELECT ROW_NUMBER() OVER () > 1 FROM test

SELECT ROW_NUMBER() OVER () BETWEEN  1 AND 10 FROM test
```
maybe nobody write sql like this, but it just works.

I copied these lines from `sqlFunctionExpression` in case they should be needed.